### PR TITLE
dlab-39: create python library not executable

### PR DESCRIPTION
**Pull request by hemeshpatel-moj**

### CODE REVIEW

File: data-products/_example_2/application/transform.py [hunk 1]
```diff
@@ -1,53 +1,32 @@
...
-# DB_USERNAME = os.environ.get("DB_USERNAME")
-# DB_PASSWORD = os.environ.get("DB_PASSWORD")
-# DB_NAME = os.environ.get("DB_NAME")
-# DB_PORT = os.environ.get("DB_PORT", 5432)
 
+def get_data(bucket: str, key: str) -> pd.DataFrame:
+    s3_client = boto3.client("s3")
+    bucket = bucket
+    key = key
+    response = s3_client.get_object(Bucket=bucket, Key=key)
```
Comment (murdo-moj) [inline data-products/_example_2/application/transform.py:11]: what happens if the key doesn't exist like that bug we saw in the uploader?

File: data-products/_example_2/application/transform.py [hunk 2]
```diff
@@ -1,53 +1,32 @@
...
 
-# This is a temporary solution, see DLAB-33 and DLAB-34 for more details.
-# DB_ENDPOINT = os.environ.get("DB_ENDPOINT")
-# DB_USERNAME = os.environ.get("DB_USERNAME")
-# DB_PASSWORD = os.environ.get("DB_PASSWORD")
-# DB_NAME = os.environ.get("DB_NAME")
-# DB_PORT = os.environ.get("DB_PORT", 5432)
 
+def get_data(bucket: str, key: str) -> pd.DataFrame:
+    s3_client = boto3.client("s3")
```
Comment (murdo-moj) [inline data-products/_example_2/application/transform.py:8]: probably good practice to instantiate the client outside of the function and pass it in as a default.

File: data-products/_example_2/package_data_product.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
 
 
 def zip_data_product(folder: str):
-    with zipfile.ZipFile(f"data_product_{folder}.zip", "w") as f:
-        for file in glob.glob(f"./{folder}/application/*") + glob.glob(f"./{folder}/requirements.txt"):
+    with zipfile.ZipFile(f"data_product{folder}.zip", "w") as f:
```
Comment (murdo-moj) [inline data-products/_example_2/package_data_product.py:6]: this is better, can you change the workflow which pushes the file so it's in line with this file naming

**Review by PriyaBasker23 — DISMISSED** [review]

File: data-products/_example_2/package_data_product.py (hunk 3 above)
Comment (PriyaBasker23) [reply data-products/_example_2/package_data_product.py:6]: I dont get what you mean by this ?

**Comment by PriyaBasker23** [comment]
All the code in replacing to get data seems ok. Just need to clarify and fix if needed with Murdo's comment

File: data-products/_example_2/package_data_product.py (hunk 3 above)
Comment (murdo-moj) [reply data-products/_example_2/package_data_product.py:6]: https://github.com/ministryofjustice/data-platform-products/blob/a6850352d992aa52a6b186972d266d752752bd7b/.github/workflows/data-product-example-2-poc.yaml#L47

**Review by murdo-moj — APPROVED** [review]

**Review by julialawrence — APPROVED** [review]
Lgtm

**Pull request merged**